### PR TITLE
chore: add `configuration` field to the PolicyReport APIs

### LINKED ISSUE #117: Add a Configuration field to PolicyReport
Some scale issues identified for `PolicyReport` can be found in this thread - https://github.com/open-policy-agent/gatekeeper/issues/2394

One option to tackle this issue is for the `PolicyReport` API to provide some kind of contract between `PolicyReport` generators and consumers. 
The proposal here is to add a `Configuration` field to the API -
```
configuration:
  limits:
    maxResults: 100
    statusFilter: ["FAIL", "WARN"]
```
This defines the max number of reports that will be stored. Also, provide a way to store only reports that the user may be interested in - the statusFilter allows to filter based on different status values - `pass, fail, error, warn, skip.`
The setting of these values should be exposed via the `PolicyReport` generator (for example, policy engines). The `PolicyReport` API should provide sane defaults for these fields

### PR DESCRIPTION
**Pull request by kumar-mallikarjuna**
**Issue**
- Closes #117 

**Changes**
- Added the `configuration` field `v1beta1` `PolicyReport` and `ClusterPolicyReport` APIs.
- Bumped the Go and Kubernetes versions.
- Updated docs.

cc: @anusha94 , @JimBugwadia

### CODE REVIEW

File: policy-report/pkg/api/wgpolicyk8s.io/v1beta1/policyreport_types.go [hunk 1]
```diff
@@ -21,6 +21,24 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// +kubebuilder:validation:Enum=pass;fail;warn;error;skip
+type StatusFilter string
+
+type Limits struct {
+	// Maximum number of results in the report
```
Comment (anusha94) [inline policy-report/pkg/api/wgpolicyk8s.io/v1beta1/policyreport_types.go:28]:
```suggestion
	// MaxResults is the maximum number of results contained in the report
```

File: policy-report/pkg/api/wgpolicyk8s.io/v1beta1/policyreport_types.go [hunk 2]
```diff
@@ -21,6 +21,24 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// +kubebuilder:validation:Enum=pass;fail;warn;error;skip
```
Comment (anusha94) [inline policy-report/pkg/api/wgpolicyk8s.io/v1beta1/policyreport_types.go:24]: Add comment:
`StatusFilter is used by PolicyReport generators to write only those reports whose status is specified by the filters`

File: policy-report/pkg/api/wgpolicyk8s.io/v1beta1/policyreport_types.go [hunk 3]
```diff
@@ -21,6 +21,24 @@ import (
...
+type StatusFilter string
+
+type Limits struct {
+	// Maximum number of results in the report
+	// +optional
+	MaxResults int `json:"maxResults"`
+
+	// Expected results list
```
Comment (anusha94) [inline policy-report/pkg/api/wgpolicyk8s.io/v1beta1/policyreport_types.go:32]:
```suggestion
	// StatusFilter indicates that the PolicyReport contains only those reports with statuses specified in this list
```

File: policy-report/pkg/api/wgpolicyk8s.io/v1beta1/clusterpolicyreport_types.go [hunk 4]
```diff
@@ -53,6 +53,11 @@ type ClusterPolicyReport struct {
 	// +optional
 	ScopeSelector *metav1.LabelSelector `json:"scopeSelector,omitempty"`
 
+	// Configuration is an optional field which can be used to specify
+	// a contract between PolicyReport generators and consumers
+	// +optional
+	Configuration *PolicyReportConfiguration `json:"configuration,omitempty"`
```
Comment (anusha94) [inline policy-report/pkg/api/wgpolicyk8s.io/v1beta1/clusterpolicyreport_types.go:59]: @JimBugwadia  Do we want to add this to `v1alpha1`?

**Comment by anusha94** [comment]
@kumar-mallikarjuna Can you also add this new field example to this file?
https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/samples/sample-v1beta1-kyverno.yaml

**Review by JimBugwadia — APPROVED** [review]

**Pull request merged**